# Mid-test: Python Use Case (Titanic Case)





##### Zumarsiyah Mahsyari
##### Astra Data Scientist Bootcamp

**Dataset** : Titanic dataset. 

You can download it here.

https://drive.google.com/open?id=15IyT1ODuDKgZb8WN6iG64hFJRZWJTSAz

**Background** : The dataset contains data for 887 of the real Titanic passengers. Each row represents one person. The columns describe different attributes about the person including whether they survived (S), their age (A), their passenger-class (C), their sex (G) and the fare they paid (X). The table below shows the Data Dictionary.

**Objective** : Create a python script that contains the script to answer the questions below. Please put comments on the script to show which line of codes answers which questions. Please upload the script after you finish.

**Questions** :
1. What is the dimension (col, row) of the data frame?
2. How to know data type of each variable?
3. How many passengers survived (Survived=1) and not-survived (Survived=0)?
4. How to drop column ‘Name’ from the data frame?
5. Add one new column called ‘family’ to represent number of family-member aboard (hint: family = sibsp + parch)
6. As shown, columns ‘Age’ contains missing values. Please add new column named ‘Age_miss’ to indicate whether Age is missing or not (Age_miss = ‘YES’ for missing value and ‘NO’ for non-missing value). 
7. Please fill Age missing value with means of existing Age values
8. What is the maximum passenger Age who survived from the tragedy? 
9. How many passengers survived from each ‘PClass’? 
10. How to randomly split the data frame into 2 parts (titanic1 and titanic2) with proportion of 0.7 for titanic1 and 0.3 for titanic2 ?

### What is the dimension (col, row) of the data frame?

In [1]:
import numpy as np
import pandas as pd

First, we need to load and read the data.

In [2]:
df = pd.read_csv("C:/Users/zumars064124/Documents/ADSB 2/Midtest/titanic.csv")
df.head()

Survived  Pclass                                               Name  \
0         0       3                             Mr. Owen Harris Braund   
1         1       1  Mrs. John Bradley (Florence Briggs Thayer) Cum...   
2         1       3                              Miss. Laina Heikkinen   
3         1       1        Mrs. Jacques Heath (Lily May Peel) Futrelle   
4         0       3                            Mr. William Henry Allen   

      Sex   Age  Siblings/Spouses Aboard  Parents/Children Aboard     Fare  
0    male  22.0                        1                        0   7.2500  
1  female  38.0                        1                        0  71.2833  
2  female  26.0                        0                        0   7.9250  
3  female  35.0                        1                        0  53.1000  
4    male  35.0                        0                        0   8.0500

In [3]:
df.shape

(887, 8)

The dimension of the dataframe is (887, 8).

### How to know data type of each variable?

Data type of each variable can be known by df.info()

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 887 entries, 0 to 886
Data columns (total 8 columns):
Survived                   887 non-null int64
Pclass                     887 non-null int64
Name                       887 non-null object
Sex                        887 non-null object
Age                        887 non-null float64
Siblings/Spouses Aboard    887 non-null int64
Parents/Children Aboard    887 non-null int64
Fare                       887 non-null float64
dtypes: float64(2), int64(4), object(2)
memory usage: 55.5+ KB


Based on the above results, it is known that Survived, Pclass, Siblings / Spouses Aboard, Parents / Children Aboard have integer data types; Age and Fare have Float data types; whereas Name and Sex have category data types.

### How many passengers survived (Survived=1) and not-survived (Survived=0)?

The number of passengers based on survived status can be determined by df['Survived'].Value_counts()

In [5]:
df['Survived'].value_counts()

0    545
1    342
Name: Survived, dtype: int64

Based on the above results, it was found that the number of passengers that survived was 342 people and the number of passengers who did not survive was 545 people.

 ### How to drop column ‘Name’ from the data frame?

Drop column 'Name' from the data frame can be done in the following way.

In [6]:
df = df.drop('Name', axis=1)
df.head()

Survived  Pclass     Sex   Age  Siblings/Spouses Aboard  \
0         0       3    male  22.0                        1   
1         1       1  female  38.0                        1   
2         1       3  female  26.0                        0   
3         1       1  female  35.0                        1   
4         0       3    male  35.0                        0   

   Parents/Children Aboard     Fare  
0                        0   7.2500  
1                        0  71.2833  
2                        0   7.9250  
3                        0  53.1000  
4                        0   8.0500

The column 'name' is no longer in the dataframe above.

### Add one new column called ‘family’ to represent number of family-member aboard (hint: family = sibsp + parch)

The column called 'Family' can be defined by adding the 'Siblings/Spouses Aboard' column and 'Parents/Children Aboard' columns

In [7]:
df['Family'] = df['Siblings/Spouses Aboard'] + df['Parents/Children Aboard']
df.head()

Survived  Pclass     Sex   Age  Siblings/Spouses Aboard  \
0         0       3    male  22.0                        1   
1         1       1  female  38.0                        1   
2         1       3  female  26.0                        0   
3         1       1  female  35.0                        1   
4         0       3    male  35.0                        0   

   Parents/Children Aboard     Fare  Family  
0                        0   7.2500       1  
1                        0  71.2833       1  
2                        0   7.9250       0  
3                        0  53.1000       1  
4                        0   8.0500       0

The 'Family' column representing the number of family-member aboard has been added to the dataframe above.

### As shown, columns ‘Age’ contains missing values. Please add new column named ‘Age_miss’ to indicate whether Age is missing or not (Age_miss = ‘YES’ for missing value and ‘NO’ for nonmissing value)

Add column 'age_miss' to the dataframe can be done in the following way.

In [25]:
df['Age_miss'] = np.where(df.Age.isnull(), 'YES', 'NO')
df.head()

Survived  Pclass     Sex   Age  Siblings/Spouses Aboard  \
0         0       3    male  22.0                        1   
1         1       1  female  38.0                        1   
2         1       3  female  26.0                        0   
3         1       1  female  35.0                        1   
4         0       3    male  35.0                        0   

   Parents/Children Aboard     Fare  Family Age_miss  
0                        0   7.2500       1       NO  
1                        0  71.2833       1       NO  
2                        0   7.9250       0       NO  
3                        0  53.1000       1       NO  
4                        0   8.0500       0       NO

The 'Age_miss' column has been added to the dataframe above. It indicates whether Age is missing or not.

### Please fill Age missing value with means of existing Age values

Fill Age missing value with means of existing Age values can be done in the following way.

In [26]:
mmean = ['Age']
for feature in mmean:
    if feature in df.columns.values:
        df[feature] = df[feature].fillna(df[feature].mean())

Filling in the missing value has been carried out so that there should be no more missing values indicated in the 'Age' column.

### What is the maximum passenger Age who survived from the tragedy? 

To find out the maximum passenger age who survived from the tragedy, we have to filter the dataframe first based on passenger who survived.

In [29]:
Survived = df[df.Survived == 1]
Ages_max = Survived['Age'].max()
print("The maximum passenger Age who survived from the tragedy is " + str(Ages_max) + ".")

The maximum passenger Age who survived from the tragedy is 80.0.


 ### How many passengers survived from each ‘PClass’?

The number of passengers survived from each 'PClass' can be determined by counting the number of passengers using value_counts () after filtering data based on passengers survived.

In [31]:
Survived['Pclass'].value_counts()

1    136
3    119
2     87
Name: Pclass, dtype: int64

Based on the above results, it is known that the results of the number of passengers who survived in PClass 1 were 136, PClass 2 was 87, and PClass 3 as many as 119.

### How to randomly split the data frame into 2 parts (titanic1 and titanic2) with proportion of 0.7 for titanic1 and 0.3 for titanic2 ? 

Randomly split the data frame into 2 part can be done in the following way.

In [38]:
titanic1 = df.sample(frac=0.7, random_state=0)
titanic2 = df.drop(titanic1.index)

Data has been divided into 2 parts, namely titanic1 by 70% and titanic2 by 30%. The following are the dimensions of each data.

In [42]:
print(titanic1.shape)
print(titanic2.shape)

(621, 9)
(266, 9)


The dimension of the titanic1 is (621, 9) and the dimension of the titanic2 is (266, 9).